### PR TITLE
Update System.Linq.Expressions to 4.1.0 to match project.json

### LINKED ISSUE #894: .Net Standard pre-release package forces downgrade of System.Linq.Expressions
As reported by @jchannon in https://twitter.com/jchannon/status/784307755305672704:

image

I see our [project.json](https://github.com/FakeItEasy/FakeItEasy/blob/c8e3b690920ec5003a6e41272d59ca01c2176b91/src/FakeItEasy.netstd/project.json) requires 4.1.0, but the [nuspec](https://github.com/FakeItEasy/FakeItEasy/blob/c8e3b690920ec5003a6e41272d59ca01c2176b91/src/FakeItEasy.netstd/FakeItEasy.nuspec) requires 4.0.11.

I think the .NET dependencies can be removed from the nuspec, but I can't find confirming documentation. Giving it a try.

### PR DESCRIPTION
**Pull request by blairconrad**
Fixes #894.

### CODE REVIEW

File: src/FakeItEasy.netstd/project.json [hunk 1]
```diff
@@ -2,21 +2,11 @@
   "dependencies": {
     "Castle.Core": "4.0.0-beta001",
     "Microsoft.Extensions.DependencyModel": "1.0.0",
+    "NETStandard.Library":"1.6.0",
```
Comment (thomaslevesque) [inline src/FakeItEasy.netstd/project.json]: Won't that cause NuGet to fetch far more dependencies than necessary? We only need a small part of what's in NETStandard.Library

Comment (blairconrad) [inline src/FakeItEasy.netstd/project.json]: Possibly. I was going to look at trimming, but first wanted to quick get on the standard versions to see if we could at least removed the version mismatch.

**Comment by jeremymeng** [comment]
I guess the fix should in the nuspec.

**Comment by jeremymeng** [comment]
@blairconrad I was thinking of bumping the version of `System.Linq.Expressions` to 4.1.0.  I am not sure nuget can resolve transitive dependency closure from the nuspec.

**Comment by blairconrad** [comment]
@jeremymeng, thanks for chiming in. I made an empty lib project and tried it out and it _seems_ to work. I created an empty lib package and restored it:

<details>
<summary>`dotnet restore` command, showing System.Linq.Expressions 4.1.0 being installed</summary>
```powershell
C:\PortableApps\Documents\Source\TestFakeItEasyDeps> dotnet restore -s ..\local-nuget --packages packages --fallbacksource https://api.nuget.org/v3/index.json
log  : Restoring packages for C:\PortableApps\Documents\Source\TestFakeItEasyDeps\project.json...
log  : Installing System.Threading.Tasks.Extensions 4.0.0.
log  : Installing System.ComponentModel 4.0.1.
log  : Installing System.ComponentModel.Primitives 4.1.0.
log  : Installing runtime.native.System 4.0.0.
log  : Installing System.Collections.NonGeneric 4.0.1.
log  : Installing System.Globalization.Extensions 4.0.1.
log  : Installing System.Runtime.Handles 4.0.1.
log  : Installing System.IO.FileSystem.Primitives 4.0.1.
log  : Installing System.Runtime.InteropServices.RuntimeInformation 4.0.0.
log  : Installing Microsoft.CSharp 4.0.1.
log  : Installing System.Xml.ReaderWriter 4.0.11.
log  : Installing System.Xml.XDocument 4.0.11.
log  : Installing System.Runtime.Serialization.Primitives 4.1.1.
log  : Installing System.Text.Encoding.Extensions 4.0.11.
log  : Installing System.Text.RegularExpressions 4.1.0.
log  : Installing System.Linq.Expressions 4.1.0.
log  : Installing FakeItEasy 2.4.0-netstd-build000034.
log  : Installing System.Runtime.Loader 4.0.0.
log  : Installing Microsoft.Extensions.DependencyModel 1.0.0.
log  : Installing Castle.Core 4.0.0-beta001.
log  : Installing System.Reflection 4.1.0.
log  : Installing System.Reflection.Emit.ILGeneration 4.0.1.
log  : Installing System.IO 4.1.0.
log  : Installing System.Runtime 4.1.0.
log  : Installing System.Diagnostics.Debug 4.0.11.
log  : Installing System.Linq 4.1.0.
log  : Installing System.Dynamic.Runtime 4.0.11.
log  : Installing Newtonsoft.Json 9.0.1.
log  : Installing Microsoft.DotNet.InternalAbstractions 1.0.0.
log  : Installing System.AppContext 4.1.0.
log  : Installing System.IO.FileSystem 4.0.1.
log  : Installing System.Reflection.TypeExtensions 4.1.0.
log  : Installing System.Resources.ResourceManager 4.0.1.
log  : Installing System.Reflection.Extensions 4.0.1.
log  : Installing System.Diagnostics.Tools 4.0.1.
log  : Installing System.Console 4.0.0.
log  : Installing System.Reflection.Emit.Lightweight 4.0.1.
log  : Installing System.Xml.XmlDocument 4.0.1.
log  : Installing System.Reflection.Emit 4.0.1.
log  : Installing System.Collections.Specialized 4.0.1.
log  : Installing System.Diagnostics.TraceSource 4.0.0.
log  : Installing System.Threading 4.0.11.
log  : Installing System.Collections 4.0.11.
log  : Installing System.ObjectModel 4.0.12.
log  : Installing System.ComponentModel.TypeConverter 4.1.0.
log  : Installing System.Runtime.Extensions 4.1.0.
log  : Installing System.Globalization 4.0.11.
log  : Installing System.Xml.XmlSerializer 4.0.11.
log  : Installing System.Runtime.InteropServices 4.1.0.
log  : Installing Microsoft.NETCore.Platforms 1.0.1.
log  : Installing Microsoft.NETCore.Targets 1.0.1.
log  : Installing System.Reflection.Primitives 4.0.1.
log  : Installing System.Text.Encoding 4.0.11.
log  : Installing System.Threading.Tasks 4.0.11.
log  : Writing lock file to disk. Path: C:\PortableApps\Documents\Source\TestFakeItEasyDeps\project.lock.json
log  : C:\PortableApps\Documents\Source\TestFakeItEasyDeps\project.json
log  : Restore completed in 55877ms.
```
</details>

And it built, too!

Here's the `project.json`.

```js
{
  "version": "1.0.0-*",
  "buildOptions": {
    "debugType": "portable"
  },
  "dependencies": {},
  "frameworks": {
    "netstandard1.6": {
      "dependencies": {
        "FakeItEasy": "2.4.0-netstd-build000034"
      }
    }
  }
}
```

Have I done something wrong?

**Comment by jeremymeng** [comment]
@blairconrad I was speculating. Your testing shows that transitive dependencies got restored soI was wrong.

I noticed that one dependency isn't brought in indirectly: `System.Collections.Concurrent`.  Is it used in the FakeItEasy source code?

**Comment by blairconrad** [comment]
Thanks, @jeremymeng. Good of you to notice the System.Collections.Concurrent difference.
FakeItEasy does use System.Collections.Concurrent. I'm not sure what to make of it not being brought in.

Perhaps every other package was pulled in via transitive dependencies listed in the nuspecs of Castle.Core, Microsoft.Extensions.DependencyModel, and System.Runtime.Loader.

**Comment by blairconrad** [comment]
After trying @jchannon's suggestion of building from the project.json and failing (no DLLs) I decided to take the quick, safe route: 
@jeremymeng's advice to just update the version of System.Collections.Concurrent in the nupkg.
Maybe once the tooling settles down we can revisit the packing from project.json.

**Comment by thomaslevesque** [comment]
> @jeremymeng's advice to just update the version of System.Collections.Concurrent in the nupkg.

Did you mean System.Linq.Expressions? I don't see any change about System.Collections.Concurrent

**Comment by blairconrad** [comment]
@thomaslevesque, I did. Thanks. My mind was on @jeremymeng's find that my other change would've omitted System.Collections.Concurrent. Sorry/thanks.

**Comment by jeremymeng** [comment]
> After trying @jchannon's suggestion of building from the project.json and failing (no DLLs)

We are using project.json only for nuget package management (like packages.config), instead of a .csproj replacement so building and packing from project.json won't work.

**Comment by blairconrad** [comment]
Hey, all. Not to be pushy or anything, but is there an outstanding action I've missed, or is this okay to merge?

**Comment by thomaslevesque** [comment]
Ah, sorry, I totally forgot about this. Will merge now.

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque. I had also forgotten about it until I opened GitKraken today and saw an unmerge branch!

**Pull request merged**